lib/streamlit/runtime/scriptrunner/script_runner.py

```diff
@@ -471,6 +472,11 @@ def _run_script(self, rerun_data: RerunData) -> None:
 
             code = self._script_cache.get_bytecode(script_path)
 
+            user = UserInfoProxy()
+            email = user.get("email", "anonymous")
+            ip = user.get("ip", "unknown ip")
+            _LOGGER.info(f"{email} [{ip}] access script {script_path}")
+
```

Comment:
I think we'll want to keep the current behavior where this type of information isn't logged by default so that people that don't expect this don't see the logs produced by their apps increase considerably. People that do want IPs/emails logged can then turn this on, which shouldn't be much of a hassle since they'll often be setting the right headers in their config already.

Reply:
Added an option `server.logPageAccess` to control wether enable logging.